galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/ras/RasSearchCriteriaStatus.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.framework.spi.ras;
+
+import javax.validation.constraints.NotNull;
+
+import dev.galasa.framework.spi.teststructure.TestStructure;
+
+public class RasSearchCriteriaStatus implements IRasSearchCriteria {
+   
+   private final String[] statuses;
+   
+   public RasSearchCriteriaStatus(@NotNull String... statuses) {
+      this.statuses = statuses;
+   }
+   
+   @Override
+   public boolean criteriaMatched(@NotNull TestStructure structure) {
+      
+      if(structure == null) {
+         return Boolean.FALSE;
+      }
+      
+      if(statuses != null) {
+         for(String status : statuses) {
+             if(status.equals(structure.getStatus())){
+                 return Boolean.TRUE;
+             }
+         }
+     }
+      
+      return Boolean.FALSE;
```

Comment:
same as above.

Reply:
Wondering where you got the inspiration for that from, as it looks wrong to me.

Reply:
Will raise a separate PR for those.